common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -655,6 +744,156 @@ public void registerExecutor(String appId, ExecutorShuffleInfo executorInfo) {
+  private void reloadActiveAppAttemptsPathInfo(DB db) throws IOException {
+    if (db != null) {
+      DBIterator itr = db.iterator();
+      itr.seek(APP_ATTEMPT_PATH_KEY_PREFIX.getBytes(StandardCharsets.UTF_8));
+      while (itr.hasNext()) {
+        Map.Entry<byte[], byte[]> e = itr.next();
+        String key = new String(e.getKey(), StandardCharsets.UTF_8);
+        if (!key.startsWith(APP_ATTEMPT_PATH_KEY_PREFIX)) {
+          break;
+        }
+        AppAttemptId appAttemptId = parseDbAppAttemptPathsKey(key);
+        try{
+          AppPathsInfo appPathsInfo = mapper.readValue(e.getValue(), AppPathsInfo.class);
+          logger.info("Reloading active application {}_{} merged shuffle files paths",
+              appAttemptId.appId, appAttemptId.attemptId);
+          appsShuffleInfo.compute(appAttemptId.appId,
+              (appId, existingAppShuffleInfo) -> {
+                if (existingAppShuffleInfo == null ||
+                    existingAppShuffleInfo.attemptId < appAttemptId.attemptId) {
+                  return new AppShuffleInfo(
+                      appAttemptId.appId, appAttemptId.attemptId, appPathsInfo);
+                } else {
+                  return existingAppShuffleInfo;
+                }
+              });
+        } catch (Exception exception) {
+          logger.error("Parsing exception is {}", exception);
+        }
+      }
+    }
+  }
+
+  private void reloadFinalizedAppAttemptsShuffleMergeInfo(DB db) throws IOException {
+    if (db != null) {
+      DBIterator itr = db.iterator();
+      itr.seek(APP_ATTEMPT_SHUFFLE_FINALIZE_STATUS_KEY_PREFIX.getBytes(StandardCharsets.UTF_8));
+      while (itr.hasNext()) {
+        Map.Entry<byte[], byte[]> e = itr.next();
+        String key = new String(e.getKey(), StandardCharsets.UTF_8);
+        if (!key.startsWith(APP_ATTEMPT_SHUFFLE_FINALIZE_STATUS_KEY_PREFIX)) {
+          break;
+        }
+        AppAttemptShuffleMergeId partitionId =
+          parseDbAppAttemptShufflePartitionKey(
+            key, APP_ATTEMPT_SHUFFLE_FINALIZE_STATUS_KEY_PREFIX);
+        logger.info("Reloading finalized shuffle info for partitionId {}",
+            partitionId.toString());
+        AppShuffleInfo appShuffleInfo = appsShuffleInfo.get(partitionId.appId);
+        if (appShuffleInfo != null && appShuffleInfo.attemptId == partitionId.attemptId) {
+          appShuffleInfo.shuffles.compute(partitionId.shuffleId,
+              (shuffleId, existingMergePartitionInfo) -> {
+                if (existingMergePartitionInfo == null ||
+                    existingMergePartitionInfo.shuffleMergeId < partitionId.shuffleMergeId) {
+                  return new AppShuffleMergePartitionsInfo(partitionId.shuffleMergeId, true);
+                } else {
+                  return existingMergePartitionInfo;
+                }
+              });
+        } else {
+          logger.warn("Removed dangling keys in DB: {}", key);
+          db.delete(e.getKey());
+        }
+      }
+    }
```

Comment:
We are not swallowing exceptions here unlike `reloadActiveAppAttemptsPathInfo`. Exception handling for these 2 should be consistent

Reply:
Refactored on exception handling